api/location_management.py

```diff
@@ -41,3 +43,48 @@ def get_schedule_data(location, date):
                 location_param  = get_id(location), 
                 date_param      = date)
             ).json()['Result']
+
+def get_event_data(restaurant: str) -> dict:
+    '''
+    Given a valid location and date,
+    perform get request, then parse the HTML code for the event_json using BeautifulSoup 4
+    '''
+    url = 'https://uci.campusdish.com/LocationsAndMenus/'
+    if(restaurant == 'Anteatery'):
+        url += 'TheAnteatery'
+    else:
+        url += restaurant
+    html = requests.get(url).text
+
+    soup = bs(html, 'html.parser')
+    table = soup.findAll('td', {'style': ['padding: 0.75pt; border-width: 0.75pt; border-right-style: dashed; border-right-color: #999999; border-bottom-style: dashed; border-bottom-color: #999999; text-align: left;',\
+                                        'padding: 0.75pt; border-width: 0.75pt; border-right-style: dashed; border-right-color: #999999; border-bottom-style: dashed; border-bottom-color: #999999; text-align: center;',\
+                                        'border-right: 0.75pt dashed #999999; border-bottom: 0.75pt dashed #999999; text-align: left;',\
+                                        'padding: 0.75pt; border-right: 0.75pt dashed #999999; text-align: left;',\
+                                        'padding: 0.75pt; border-bottom: 0.75pt dashed #999999; text-align: left;']})
+    entries_list = list()
+
+    for i in table:
+        item_text = i.getText().strip('\n')
+        entries_list.append(item_text)
+
+    headers = ('date', 'name', 'service_start', 'service_end')
+    rows_list = entries_list[5:]
+
+    row = {}
+    curr_time = get_irvine_time()
+    for i in range(0, int(len(rows_list) / 4)):
```

Comment:
What does this loop do? I would think we wouldn't need a loop here as currently the function only returns one row.

Reply:
It goes through all the event data to filter out the one closest to current time